### DISCUSSION TITLE
How about non-stochastic meteorological variables?

### DISCUSSION THREAD

**Discussion by muhshif** [btschwertfeger/python-cmethods#22] [Q&A]
As I can see as the example in the documentation, you demonstrated the CMethods by using temperature which is a non-stochastic variable (as of my understanding). But, whether we can apply these methods to rainfall (or precipitation) which is a stochastic variable?
[I hope you understand why atmospheric science researchers always consider rainfall as a stochastic variable. And in most of the statistical analysis therefore rainfall gets another treatment unlike temperature, relative humidity, etc.]

**Comment by btschwertfeger** [comment]
In the documentation of the python-cmethods package, climate variables are classified into two types: "interval-based" and "ratio-based". In this context, "interval-based" refers to non-stochastic variables such as air temperatures, while "ratio-based" variables are considered stochastic.

The bias correction methods provided in the python-cmethods package are applicable to both stochastic and non-stochastic variables, with the exception of variance scaling. When calling a method, the parameter "kind" can be specified, which is set to "+" (additive) by default for all methods. Additive methods are suitable for non-stochastic variables. However, to adjust precipitation or air pressure, the parameter "kind" must be set to "*" (multiplicative).

The following example shows how to apply the linear scaling procedure on a non-stochastic climate variable (works at python-cmethods v1.0.0):
```python
import xarray as xr
from cmethods import CMethods as cm

# Note: The data sets must contain the dimension "time"
#       for the respective variable.
obsh = xr.open_dataset("path/to/reference_data-control_period.nc")
simh = xr.open_dataset("path/to/modeled_data-control_period.nc")
simp = xr.open_dataset("path/to/the_dataset_to_adjust-scenario_period.nc")
variable = "tas" # temperatures

qdm_adjusted = cm.linear_scaling(
     obs=obs[variable],
     simh=simh[variable],
     simp=simp[variable],
     kind="+" # <- change to "*" for adjusting stochastic climate variables.
)
```


The "kind" parameter can be selected for all available methods, including Linear Scaling (LS), Variance Scaling (VS), Delta (change) Method (DM), Quantile Mapping (DQM), Detrended Quantile Mapping (QM), and Quantile Delta Mapping (QDM). The additive method is available for all methods, while the multiplicative method is not available for variance scaling.The formulas for these methods can be found in the documentation here: https://python-cmethods.readthedocs.io/en/stable/src/cmethods.html#cmethods.CMethods.
